transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollSocketChannel.java

```diff
@@ -110,6 +114,33 @@ protected AbstractEpollUnsafe newUnsafe() {
         return new EpollSocketChannelUnsafe();
     }
 
+    @Override
+    boolean doConnect0(SocketAddress remote) throws Exception {
+        if (Native.IS_SUPPORTING_TCP_FASTOPEN && config.isTcpFastOpenConnect()) {
+            pendingFastOpenWrite = true;
+            return true;
```

Comment:
Should this return false?   The parent caller in `AbstractEpollChannel.doConnect` seems to expect the remote address to be set, even though it won't be.

Reply:
If I return false here the tests end up hanging waiting for the connect to complete. Hmm still trying to figure out exactly how the logic works in `AbstractEpollChannel.doConnect` for setting remote, but it does seem to get set using the provided remoteAddress using (non-hostname) IPs in the tests on Java 8. That behavior is still required as we need the channel's remoteAddress set so we can do the stateless connect in the first write call.

Reply:
confirmed that AbstractEpollChannel.doConnect will still set remote using the provided address always if the address is missing on the socket. hmmm wondering how this may effect addresses with hostnames